File: backend/src/main/java/newstock/domain/stock/entity/Stock.java
package newstock.domain.stock.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Stock {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer stockId;

    private String stockCode;

    private String stockName;

    private Integer closingPrice;

    private String rcPdcp;

    private String imgUrl;

    private Integer totalPrice;

    private String capital;

    private String lstgStqt;

    private String listingDate;

    private String stdIccn;

}

File: frontend/types/api/stock.ts
import { BaseResType } from './base';

export type AllStockResType = BaseResType & {
  data: {
    stockList: {
      stockId: number;
      stockCode: string;
      stockName: string;
      isInterested: boolean;
      imgUrl: string;
    }[];
  };
};

export type StockInterestReqType = {
  stockIdList: number[];
};

export type StockInterestResType = BaseResType;

export type AllUserStockResType = BaseResType & {
  data: {
    stockList: StockType[];
  };
};

export type StockDetailInfoResType = BaseResType & {
  data: {
    stockId: number;
    stockName: string;
    closingPrice: number;
    rcPdcp: number;
    stockImage: string;
    totalPrice: string;
    capital: string;
    lstgStqt: string;
    parValue: string;
    issuePrice: string;
    listingDate: string;
    stdIccn: string;
    ctpdPrice: number;
  };
};

export type StockType = {
  stockId: number;
  stockCode: string;
  stockName: string;
  closingPrice: number;
  rcPdcp: number;
  imgUrl: string;
};
